src/main/java/core/basesyntax/strategy/mapvalidator/KeyAbsentValidator.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.mapvalidator;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public interface KeyAbsentValidator extends MapValidator {
+    @Override
+    default void validateMap(Map<String, Integer> fruitQuantityMap,
+                                    FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        if (fruitQuantityMap.containsKey(fruit)) {
+            String exceptionMessage = "Fruit from transaction = ["
+                    + transaction + "] was already present in the map in quantity of = ["
+                    + fruitQuantityMap.get(fruit) + "]";
+            throw new IllegalArgumentException(exceptionMessage);
+        }
+    }
```

Comment:
According to the checklist, avoid using if-else constructions when applying the Strategy pattern. Consider refactoring this code to adhere to the Strategy pattern more cleanly.